scipy/stats/_distn_infrastructure.py

```diff
@@ -3913,6 +3911,175 @@ def _get_support(self, *args):
         """
         return self.a, self.b
 
+    def _pmf(self, x):
+        return rv_count._pmf(self, x)
+
+    def _cdf(self, x):
+        return rv_count._cdf(self, x)
+
+    def _ppf(self, q):
+        return rv_count._ppf(self, q)
+
+    def _rvs(self, size=None, random_state=None):
+        return rv_count._rvs(self, size, random_state)
+
+    def _entropy(self):
+        return rv_count._entropy(self)
+
+    def generic_moment(self, n):
+        return rv_count.generic_moment(self, n)
+
+    def _expect(self, fun, lb, ub, *args, **kwds):
+        return rv_count._expect(self, fun, lb, ub, *args, **kwds)
+
+
+class rv_count(rv_discrete):
+    """A discrete random variable defined by support points and probabilities.
+
+    Parameters
+    ----------
+    xk : array_like
+        Support points.
+    pk : array_like
+        The probabilities, corresponding to `xk`
+    name : str, optional
+        The name of the instance. This string is used to construct the default
+        example for distributions.
+    longname : str, optional
+        This string is used as part of the first line of the docstring returned
+        when a subclass has no docstring of its own. Note: `longname` exists
+        for backwards compatibility, do not use for new subclasses.
```

Comment:
Do we still need these (especially `longname`)? If so, can we accept them as `kwargs` and not document them?

Reply:
A short `name` is somewhat useful I think. E.g. `rv_count(..., name='cat count')` and `rv_count(..., name='dog count')`.  `longname` may be indeed there for backwards compat only, so can in principle remove. Going through kwargs however adds possible user confusion (what other kwargs are they accepting and what they mean?) and a burden of validating them. So I'd rather keep this as is.